android/app/src/main/java/org/fossasia/openevent/viewmodels/FeedFragmentViewModel.java

```diff
@@ -0,0 +1,55 @@
+package org.fossasia.openevent.viewmodels;
+
+import android.arch.lifecycle.ViewModel;
+import android.content.res.Resources;
+
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.api.APIClient;
+import org.fossasia.openevent.data.facebook.FeedItem;
+import org.fossasia.openevent.utils.ConstantStrings;
+import org.fossasia.openevent.utils.SharedPreferencesUtil;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.schedulers.Schedulers;
+
+public class FeedFragmentViewModel extends ViewModel {
+
+    private List<FeedItem> feedLive;
+
+    public FeedFragmentViewModel() {
+        feedLive = new ArrayList<>();
+    }
+
+    public List<FeedItem> getFeedList() {
+        return feedLive;
+    }
+
+    public void clearFeedList() {
+        feedLive.clear();
+    }
+
+    public void addAllFeed(List<FeedItem> feed) {
+        feedLive.addAll(feed);
+    }
+
+    public void updateFBPageID(Resources resources) {
+        if (SharedPreferencesUtil.getString(ConstantStrings.FACEBOOK_PAGE_ID, null) == null)
+            APIClient.getFacebookGraphAPI().getPageId(SharedPreferencesUtil.getString(ConstantStrings.FACEBOOK_PAGE_NAME, null),
+                    resources.getString(R.string.facebook_access_token))
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe(facebookPageId -> {
+                        String id = facebookPageId.getId();
+                        SharedPreferencesUtil.putString(ConstantStrings.FACEBOOK_PAGE_ID, id);
+                    });
```

Comment:
Convert this Observable to LiveData and return it to fragment and allow Fragment to handle its result